Sources/Features/Breadcrumb/BacktraceComponentListener.swift

```diff
@@ -0,0 +1,126 @@
+                print("unknown")
+        }
+    }
+    
+    private func addBradcrumb(_ orientation: String) {
+        let attributes = ["orientation": orientation]
+        let _ = BacktraceClient.shared?.addBreadcrumb("Configuration changed",
+                                                      attributes: attributes,
+                                                      type: .system,
+                                                      level: .info)
+    }
+    
+    
+//MARK: Memroy Status Listener
+    
+    @objc private func notifyMemoryStatusChange() {
+        let _ = BacktraceClient.shared?.addBreadcrumb("Critical low memory warning!",
+                                                      type: .system,
+                                                      level: .fatal)
+    }
+    
+    private var source: DispatchSourceMemoryPressure?
+    @objc private func observeMemoryStatusChanged() {
+        NotificationCenter.default.addObserver(self, selector: #selector(notifyMemoryStatusChange), name: UIApplication.didReceiveMemoryWarningNotification, object: nil)
+        
+        if let source:DispatchSourceMemoryPressure = DispatchSource.makeMemoryPressureSource(eventMask: .all, queue:DispatchQueue.main) as? DispatchSource {
+            let eventHandler: DispatchSourceProtocol.DispatchSourceHandler = {
+                let event:DispatchSource.MemoryPressureEvent = source.data
+                if source.isCancelled == false {
+                    self.didReceive(event)
+                }
+            }
+            source.setEventHandler(handler:eventHandler)
+            source.setRegistrationHandler(handler:eventHandler)
+            source.activate()
+            self.source = source
+        }
+    }
+    
+    private func getMemoryWarningText(_ memoryPressureEvent: DispatchSource.MemoryPressureEvent) -> String {
+        if memoryPressureEvent.rawValue == DispatchSource.MemoryPressureEvent.normal.rawValue {
+            return "Generic memory warning"
+        } else if memoryPressureEvent.rawValue == DispatchSource.MemoryPressureEvent.critical.rawValue {
+            return "TRIM MEMORY RUNNING CRITICAL"
+        } else if memoryPressureEvent.rawValue == DispatchSource.MemoryPressureEvent.warning.rawValue {
+            return "TRIM MEMORY RUNNING WARNING"
+        } else {
+            return "Generic memory warning"
+        }
+    }
+    
+    private func getMemoryWarningLevel(_ memoryPressureEvent: DispatchSource.MemoryPressureEvent) -> BacktraceBreadcrumbLevel {
+        if memoryPressureEvent.rawValue == DispatchSource.MemoryPressureEvent.normal.rawValue {
+            return .warning
+        } else if memoryPressureEvent.rawValue == DispatchSource.MemoryPressureEvent.critical.rawValue {
+            return .fatal
+        } else if memoryPressureEvent.rawValue == DispatchSource.MemoryPressureEvent.warning.rawValue {
+            return .error
+        } else {
+            return .warning
```

Comment:
Same question as for the message, the first if and default option being the same (`.warning`).

Reply:
With else part we have to return BacktraceBreadcrumbLevel that's why choosen .warning. Please do suggest any other BacktraceBreadcrumbLevel if you have.

Reply:
`.info` or `.debug`?